extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/devmode/console/ConfigEditorProcessor.java

```diff
@@ -1,171 +1,90 @@
-                    current.getOptionalValue(i.getPropertyName(), String.class).orElse(null),
-                    appProperties.getProperty(i.getPropertyName())));
+    @BuildStep(onlyIf = IsDevelopment.class)
+    @Record(ExecutionTime.RUNTIME_INIT)
+    public DevConsoleRuntimeTemplateInfoBuildItem config(ConfigRecorder recorder,
+            List<ConfigDescriptionBuildItem> configDescriptionBuildItems) {
+        List<ConfigDescription> configDescriptions = new ArrayList<>();
+        for (ConfigDescriptionBuildItem item : configDescriptionBuildItems) {
+            configDescriptions.add(
+                    new ConfigDescription(item.getPropertyName(), item.getDocs(), item.getDefaultValue()));
         }
-        Collections.sort(configs);
-
-        return new DevConsoleTemplateInfoBuildItem("config", configs);
+        return new DevConsoleRuntimeTemplateInfoBuildItem("config", new ConfigDescriptionsSupplier(configDescriptions));
     }
 
     @BuildStep
     DevConsoleRouteBuildItem handlePost() {
         return new DevConsoleRouteBuildItem("config", "POST", new DevConsolePostHandler() {
             @Override
             protected void handlePost(RoutingContext event, MultiMap form) throws Exception {
-                String key = event.request().getFormAttribute("name");
+                String name = event.request().getFormAttribute("name");
                 String value = event.request().getFormAttribute("value");
 
-                Properties appProperties = new Properties();
-                Path appProps = null;
-                for (Path i : DevConsoleManager.getHotReplacementContext().getResourcesDir()) {
-                    Path app = i.resolve("application.properties");
-                    if (Files.exists(app)) {
-                        appProps = app;
+                List<Path> resourcesDir = DevConsoleManager.getHotReplacementContext().getResourcesDir();
+                if (resourcesDir.isEmpty()) {
+                    throw new IllegalStateException("Unable to manage configurations - no resource directory found");
+                }
+
+                // In the current project only
+                Path path = resourcesDir.get(0);
+                Path configPath = path.resolve("application.properties");
+                if (!Files.exists(configPath)) {
+                    configPath = Files.createFile(path.resolve("application.properties"));
+                }
+
+                String profile = ProfileManager.getActiveProfile();
+                name = !profile.equals(DEVELOPMENT.getDefaultProfile()) ? "%" + profile + "." + name : name;
+                List<String> lines = Files.readAllLines(configPath);
+                int nameLine = -1;
+                for (int i = 0, linesSize = lines.size(); i < linesSize; i++) {
+                    final String line = lines.get(i);
+                    if (line.startsWith(name)) {
```

Comment:
This will break for property names that are also prefixes, as you don't check for the '=' like the existing code does (e.g. anything that uses io.quarkus.runtime.annotations.ConfigItem#PARENT)